commons/common_separation.py

```diff
@@ -0,0 +1,168 @@
+
+    Returns:
+    -------
+    df : pandas DataFrame
+        The transcribed data in a DataFrame.
+    speakers: int
+        The number of speakers detected in the transcription.
+
+    ------------------------------------------------------------------------------------------------------
+    """
+    speakers = 0
+    df = pd.DataFrame()
+
+    if 'results' in response:
+        if 'speaker_labels' in response['results']:
+
+            if 'speakers' in response['results']['speaker_labels']:
+                speakers = response['results']['speaker_labels']["speakers"]
+
+            if 'items' in response['results']:
+                items = response['results']["items"]
+                df = pd.DataFrame(items)
+
+                df["confidence"] = df["alternatives"].apply(lambda x: x[0]["confidence"])
+                df["content"] = df["alternatives"].apply(lambda x: x[0]["content"])
+                df["confidence"] = df["confidence"].astype(float)
+
+                df = df[df["confidence"] > 0].reset_index(drop=True)
+                df = df[["start_time", "end_time", "confidence", "speaker_label", "content"]]
+                
+    return df, speakers
+
+def extract_data(segment_info):
+    """
+    ------------------------------------------------------------------------------------------------------
+
+    extract data from word_info
+
+    Parameters:
+    ----------
+    segment_info : object
+        The phrase level transcribed data.
+
+    Returns:
+    -------
+    df : pandas series
+        The phrase level transcribed data in a pandas series.
+
+    ------------------------------------------------------------------------------------------------------
+    """
+    words = segment_info.get("words", None)
+
+    starts = [word.get("start", np.nan) for word in words]
+    ends = [word.get("end", np.nan) for word in words]
+    phrases = [word.get("word", "") for word in words]
+    scores = [word.get("score", 0) for word in words]
+    speakers = [segment_info.get("speaker", "no_speaker") for _ in words]
+
+    return pd.DataFrame({"start": starts, "end": ends, "phrase": phrases, "score": scores, "speaker": speakers})
+
+def whisperx_to_dataframe(json_response):
+    """
+    ------------------------------------------------------------------------------------------------------
+
+    Transcribes(local:whisperx) a json response into a pandas DataFrame.
+
+    Parameters:
+    ----------
+    json_response : dict
+        The response object containing the transcribed data.
+
+    Returns:
+    -------
+    df : pandas DataFrame
+        The transcribed data in a DataFrame.
+    speakers: int
+        The number of speakers detected in the transcription.
+
+    ------------------------------------------------------------------------------------------------------
+    """
+    df = pd.DataFrame(columns=["start_time", "end_time", "content", "confidence", "speaker_label"])
```

Comment:
Just checking: are we planning to hard code variable names?

Reply:
this is not new code, I moved it from `separation_util` into commons because it is used in both `openwillis-voice` and `openwillis-transcribe` (forgot to remove it from `separation_util` before: https://github.com/bklynhlth/openwillis/pull/177/commits/fc0b903ae98ffe7c771075969ba4bd265e585a4e). In the future we can modify it to get column names from measures.